pkg/scheduler/framework/plugins/nodedeclaredfeatures/nodedeclaredfeatures_test.go

```diff
@@ -0,0 +1,521 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodedeclaredfeatures
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/mock"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/klog/v2/ktesting"
+	fwk "k8s.io/kube-scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/names"
+	"k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+	st "k8s.io/kubernetes/pkg/scheduler/testing"
+
+	ndf "k8s.io/component-helpers/nodedeclaredfeatures"
+	"k8s.io/component-helpers/nodedeclaredfeatures/features"
+	ndftesting "k8s.io/component-helpers/nodedeclaredfeatures/testing"
+)
+
+// createMockFeature is a helper function to create and configure a MockFeature.
+func createMockFeature(t *testing.T, name string, infer bool, minVersionStr string) *ndftesting.MockFeature {
+	t.Helper()
+	m := ndftesting.NewMockFeature(t)
+	m.EXPECT().Name().Return(name).Maybe()
+	m.EXPECT().InferForScheduling(mock.Anything).Return(infer)
+	minVersion := version.MustParseSemantic("1.30.0") // Default
+	if minVersionStr != "" {
+		minVersion = version.MustParseSemantic(minVersionStr)
+	}
+	m.EXPECT().MinVersion().Return(minVersion).Maybe()
+	m.EXPECT().MaxVersion().Return(nil).Maybe()
+	return m
+}
+
+func TestPreFilter(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	testCases := []struct {
+		name              string
+		pod               *v1.Pod
+		mockFeatures      []*ndftesting.MockFeature
```

Comment:
```suggestion
		features      []ndf.Feature
```
Then you can simply use `features` in the test